### PR TITLE
Taylorr/bugfix/random fixes

### PR DESCRIPTION
**Pull request by Taylorr82**
A collection of a bunch of things that were messing with other things on the system. Most noticeable was FTP because of the high activity, but there was probably some hidden problems with some of these bugs.

### CODE REVIEW

File: ex2_services/Services/source/adcs/adcs_service.c [hunk 1]
```diff
@@ -1531,6 +1531,13 @@ SAT_returnState adcs_service_app(csp_packet_t *packet) {
  * @return None
  */
 void adcs_service(void *param) {
+    int32_t iErr = red_mkdir("adcs");
+    if (iErr == -1 && red_errno != 17) {
+        sys_log(ERROR, "Unexpected error %d from creating adcs directory", red_errno);
+    } else {
+        sys_log(INFO, "Successfully created adcs directory");
+    }
+
```
Comment (JenishPatel99) [inline ex2_services/Services/source/adcs/adcs_service.c:1540]: Just wondering why do we move adcs directory creation from below to start of function?

Comment (Taylorr82) [reply ex2_services/Services/source/adcs/adcs_service.c:1540]: https://github.com/AlbertaSat/ex2_obc_software/pull/287/commits/d1ec26862ca5864651783f5b5ee739ace2c2df94
The description of this commit

File: main/main.c [hunk 2]
```diff
@@ -295,7 +295,7 @@ static void init_csp() {
     csp_conf.conn_max = 20;
     csp_conf.conn_queue_length = 10;
     csp_conf.fifo_length = 25;
-    csp_conf.port_max_bind = 24;
+    csp_conf.port_max_bind = 254;
```
Comment (JenishPatel99) [inline main/main.c:298]: What does this change do?

File: reliance_edge/include/redconf.h [hunk 3]
```diff
@@ -51,7 +51,7 @@
 
 #define REDCONF_API_FSE_TRANSMASKSET 0
 
-#define REDCONF_OUTPUT 1
+#define REDCONF_OUTPUT 0
```
Comment (JenishPatel99) [inline reliance_edge/include/redconf.h:54]: What will this change?

Comment (Taylorr82) [reply reliance_edge/include/redconf.h:54]: The output wasn't actually going anywhere but it was still being made. This is a performance improvement

File: main/main.c (hunk 2 above)
Comment (Taylorr82) [reply main/main.c:298]: We ran out of ports for our services so I increased it to the maximum

**Review by JenishPatel99 — APPROVED** [review]

File: ex2_services/Services/source/housekeeping/housekeeping_service.c [hunk 4]
```diff
@@ -573,18 +573,7 @@ Result write_hk_to_file(uint16_t filenumber, All_systems_housekeeping *all_hk_da
...
-    red_write(fout, &all_hk_data->Athena_hk, sizeof(all_hk_data->Athena_hk));
-    red_write(fout, &all_hk_data->EPS_hk, sizeof(all_hk_data->EPS_hk));
-    red_write(fout, &all_hk_data->UHF_hk, sizeof(all_hk_data->UHF_hk));
-    red_write(fout, &all_hk_data->S_band_hk, sizeof(all_hk_data->S_band_hk));
-    red_write(fout, &all_hk_data->hyperion_hk, sizeof(all_hk_data->hyperion_hk));
-    red_write(fout, &all_hk_data->charon_hk, sizeof(all_hk_data->charon_hk));
-    red_write(fout, &all_hk_data->DFGM_hk, sizeof(all_hk_data->DFGM_hk));
-    red_write(fout, &all_hk_data->NS_hk, sizeof(all_hk_data->NS_hk));
-    red_write(fout, &all_hk_data->IRIS_hk, sizeof(all_hk_data->IRIS_hk));
+    red_write(fout, all_hk_data, sizeof(All_systems_housekeeping));
```
Comment (rcunrau) [inline ex2_services/Services/source/housekeeping/housekeeping_service.c:576]: The seek used "needed_size", but the write uses sizeof(All_systems_housekeeping). Are they the same?

**Review by rcunrau — APPROVED** [review]
Back in our perfect history days we separated formatting commits from semantic commits, and prefaced the non-semantic commits with "WHITESPACE" (actually, that was probably the entire commit message). I think you're mostly keeping them separate, but we may want to incorporate that policy into our best practices or coding style document.

File: ex2_services/Services/source/housekeeping/housekeeping_service.c (hunk 4 above)
Comment (Taylorr82) [reply ex2_services/Services/source/housekeeping/housekeeping_service.c:576]: I'm not sure what that's about. Maybe I should give it a proper review

File: ex2_services/Services/source/services.c [hunk 5]
```diff
@@ -56,43 +56,32 @@ SAT_returnState start_service_server(void) {
...
+                                          "scheduler_service", "housekeeping_service",  "general_service",
+                                          "logger_service",    "dfgm_service",          "adcs_service",
+                                          "FTP_service",       "ns_payload_service",    "iris_service"};
 
-    int number_of_cmds = ((sizeof(start_service_function) - 1) / sizeof(services));
-
-    uint8_t *start_service_flag = pvPortMalloc(number_of_cmds * sizeof(uint8_t));
-    memset(start_service_flag, 0, number_of_cmds * sizeof(uint8_t));
-    int start_service_retry;
+    services start_service_function[] = {&start_cli_service,
```
[the remark this answers is not in the document]
Comment (Taylorr82) [reply ex2_services/Services/source/services.c:64]: Ah, I'll clean that up

**Comment by Taylorr82** [comment]
> Back in our perfect history days we separated formatting commits from semantic commits, and prefaced the non-semantic commits with "WHITESPACE" (actually, that was probably the entire commit message). I think you're mostly keeping them separate, but we may want to incorporate that policy into our best practices or coding style document.

I would like formatting commits separate, but with the whole pre-commit hook thing formatting changes can sneak in. I don't think the team is experienced enough to separate formatting and code commits properly

**Review by tganley — APPROVED** [review]

**Review by JenishPatel99 — APPROVED** [review]
Looks good

**Review by tganley — APPROVED** [review]

**Pull request merged**